### PR TITLE
Migrate toolcache / toolset tests to Pester

### PR DESCRIPTION
**Pull request by maxim-lobanov**
# Description
- Migrate toolcache / toolset tests to Pester
- Implement helper function `Get-WhichTool`

Test build: https://dev.azure.com/github/virtual-environments/_build/results?buildId=75385&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 1]
```diff
@@ -360,6 +360,12 @@ function Get-ToolsetContent
     ConvertFrom-Json -InputObject $toolsetJson
 }
 
+function Get-ToolcacheToolDirectory {
+    Param ([string] $ToolName)
+    $toolcacheRootPath = $env:AGENT_TOOLSDIRECTORY.Replace("/", "\")
```
Comment (al-cheb) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1:365]: nitpick: Resolve-Path

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 2]
```diff
@@ -360,6 +360,12 @@ function Get-ToolsetContent
...
+    return Join-Path $toolcacheRootPath $Name
```
Comment (al-cheb) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1:366]: Should be $ToolName?

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 (hunk 1 above)
Comment (vmapetr) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1:365]: `Resolve-Path` instead of `Replace`?

Comment (maxim-lobanov) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1]: Thanks, fixed

Comment (maxim-lobanov) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1]: Thanks, fixed

**Pull request merged**